internal/resources/azure/container_registry.go

```diff
@@ -0,0 +1,178 @@
+	if r.MonthlyBuildVCPUHrs != nil {
+		monthlyBuildVCPU = decimalPtr(decimal.NewFromFloat(*r.MonthlyBuildVCPUHrs * 3600))
+	}
+
+	costComponents = append(costComponents, ContainerRegistryCPUCostComponent("Build vCPU", region, sku, monthlyBuildVCPU))
+
+	return &schema.Resource{
+		Name:           r.Address,
+		CostComponents: costComponents, UsageSchema: ContainerRegistryUsageSchema,
+	}
+}
+
+func ContainerRegistryCostComponent(name, region, sku string) *schema.CostComponent {
+	return &schema.CostComponent{
+		Name:            name,
+		Unit:            "days",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: decimalPtr(decimal.NewFromInt(30)),
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("azure"),
+			Region:        strPtr(region),
+			Service:       strPtr("Container Registry"),
+			ProductFamily: strPtr("Containers"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "productName", Value: strPtr("Container Registry")},
+				{Key: "skuName", Value: strPtr(sku)},
+				{Key: "meterName", Value: strPtr(fmt.Sprintf("%s Registry Unit", sku))},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			PurchaseOption: strPtr("Consumption"),
+		},
+	}
+}
+func ContainerRegistryGeolocationCostComponent(name, region, sku string, locationsCount int) *schema.CostComponent {
+	return &schema.CostComponent{
+		Name:            name,
+		Unit:            "days",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: decimalPtr(decimal.NewFromInt(30 * int64(locationsCount))),
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("azure"),
+			Region:        strPtr(region),
+			Service:       strPtr("Container Registry"),
+			ProductFamily: strPtr("Containers"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "productName", Value: strPtr("Container Registry")},
+				{Key: "skuName", Value: strPtr(sku)},
+				{Key: "meterName", Value: strPtr(fmt.Sprintf("%s Registry Unit", sku))},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			PurchaseOption: strPtr("Consumption"),
+		},
+	}
+}
+func ContainerRegistryStorageCostComponent(name, region, sku string, storage *decimal.Decimal) *schema.CostComponent {
+	return &schema.CostComponent{
+
+		Name:            name,
+		Unit:            "GB",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: storage,
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("azure"),
+			Region:        strPtr(region),
+			Service:       strPtr("Container Registry"),
+			ProductFamily: strPtr("Containers"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "productName", Value: strPtr("Container Registry")},
+				{Key: "skuName", Value: strPtr(sku)},
+				{Key: "meterName", Value: strPtr("Data Stored")},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			PurchaseOption: strPtr("Consumption"),
+		},
+	}
+}
+func ContainerRegistryCPUCostComponent(name, region, sku string, monthlyBuildVCPU *decimal.Decimal) *schema.CostComponent {
```

Comment:
Should this be a private function on the struct?

Reply:
Fixed